### PR TITLE
[gitlab] remove duplicate issue lookup in link flow

### PR DESCRIPTION
**Pull request by dcramer**
This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry-plugins/28)

### CODE REVIEW

**Comment by dcramer** [comment]
this is what I was talking about yesterday -- where the updates to the link_issue flow let us remove this duplicate call

File: src/sentry_plugins/github/plugin.py [hunk 1]
```diff
@@ -160,9 +160,10 @@ def link_issue(self, request, group, form_data, **kwargs):
         comment = form_data.get('comment')
         if comment:
             try:
+
                 client.create_comment(
                     repo=self.get_option('repo', group.project),
-                    issue_id=form_data['issue_id'],
+                    global_issue_id=issue['id'],
```
Comment (macqueen) [inline src/sentry_plugins/github/plugin.py]: umm this is the github plugin...

**Comment by dcramer** [comment]
Damnit. Idk how I fucked that one up. Gonna add some tests.

File: src/sentry_plugins/github/plugin.py (hunk 1 above)
Comment (mattrobenolt) [inline src/sentry_plugins/github/plugin.py]: lol

**Comment by dcramer** [comment]
Added some tests and fixed a bunch of minor quirks. Still can probably do more here, but at the very least it'll be hard to break these two in the future.

File: tests/gitlab/test_plugin.py [hunk 2]
```diff
@@ -74,3 +71,35 @@ def test_create_issue(self):
             'labels': None,
             'assignee_id': None,
         }
+
+    @responses.activate
+    @override_settings(GITHUB_APP_ID='abc', GITHUB_API_SECRET='123')
```
Comment (macqueen) [inline tests/gitlab/test_plugin.py]: should this be gitlab or github? (also on line 48)

**Pull request merged**